### PR TITLE
Refactor thread pool to dynamic threads mode

### PR DESCRIPTION
**Pull request by zhagnlu**
#26406

### CODE REVIEW

**Comment by zhagnlu** [comment]
rerun ut

File: internal/core/src/storage/ThreadPool.h [hunk 1]
```diff
@@ -60,6 +63,12 @@ class ThreadPool {
     void
     ShutDown();
 
+    size_t
+    GetThreadNum() {
+        std::lock_guard<std::mutex> lock(mutex_);
+        return current_threads_size_;
```
Comment (czs007) [inline internal/core/src/storage/ThreadPool.h:69]: how about https://en.cppreference.com/w/cpp/thread/shared_mutex

Comment (czs007) [reply internal/core/src/storage/ThreadPool.h:69]: you can  modify this in next pr. Since this GetThreadNum is not called by uppper layer.

**Pull request merged**